File: backend/internal/models/skill.go
package models

import "time"

type Skill struct {
	ID       int    `json:"id"`
	Username string `json:"username"`
	Skill    string `json:"skill"`
	Exchange string `json:"exchange"`
	Category string `json:"category"`
}

type SkillDescription struct {
	ID          int       `json:"id"`
	SkillID     int       `json:"skill_id"`    // FK к skills
	Description string    `json:"description"` // Подробное описание
	CreatedAt   time.Time `json:"created_at"`
	Media       string    `json:"media,omitempty"`
	Skill       string    `json:"skill,omitempty"`
	Exchange    string    `json:"exchange,omitempty"`
	Username    string    `json:"username,omitempty"`
}

type SkillFull struct {
	ID          int       `json:"id"`
	Username    string    `json:"username"`
	Skill       string    `json:"skill"`
	Exchange    string    `json:"exchange"`
	Category    string    `json:"category"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	Media       string    `json:"media,omitempty"`
}

File: frontend/src/types/skill.ts
import type { categories, categoriesView } from '../constants/categories';
import type { skillStatus } from '../constants/skills';

export interface ISkill {
  id: string;
  category: CategoryViewType;
  description: string;
  skill: string;
  exchange: string;
  contactType: ContactType;
  contactValue: string | null;
  username: string;
  avatarUsername: string | null;
  createdAt: string;
}

export interface ISkillHistory extends ISkill {
  status: SkillStatusType;
}

export type IEditSkill = Omit<
  ISkill,
  'id' | 'username' | 'avatarUsername' | 'createdAt' | 'category'
> & { category: CategoryType };

export type CategoryType = (typeof categories)[number];
export type CategoryViewType = (typeof categoriesView)[number];

export type ContactType = 'site' | 'telegram' | 'vk' | 'wechat';

export type SkillHistorySearchType = SkillStatusType | 'ALL';
export type SkillStatusType = keyof typeof skillStatus;
